### PR TITLE
[backport] Backport image mirroring for testing

### PR DESCRIPTION
**Pull request by Itxaka**
Does it fix an issue? addresses a business case?

add a description and link to the issue if one exists.

Fixes SUSE/avant-garde#1831

## What does this PR do?

please include a brief "management" technical overview (details are in the code)

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by Itxaka** [comment]
This needs to be backported to 4.2 Im guessing https://github.com/SUSE/skuba/pull/1356

File: ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile [hunk 1]
```diff
@@ -89,8 +89,8 @@ node('caasp-team-private-integration') {
            }
            if (pr_repo_label != null) {
                def branch_name = pr_repo_label.name.split(":")[1]
-               branch_repo = "http://download.suse.de/ibs/Devel:/CaaSP:/4.0:/Branches:/${branch_name}/SLE_15_SP1"
-               branch_registry = "registry.suse.de/devel/caasp/4.0/branches/${branch_name}/containers"
+               branch_repo = "http://download.suse.de/ibs/Devel:/CaaSP:/${repo_version}:/Branches:/${branch_name}/SLE_15_SP1"
+               branch_registry = "registry.suse.de/Devel/CaaSP/${repo_version}/Branches/${branch_name}/containers"
```
Comment (davidcassany) [inline ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile:93]: mmm not sure why there is this change here... registry references with capital letters are not allowed... which makes me note that `${branch_name}` should turned in to lowercase characters on this string composition. In any case this is fix for master not the backport itself.

**Review by davidcassany — CHANGES_REQUESTED** [review]
The image reference URI (well not sure if image references are URI compliant) is not correct. Uppercase characters are not allowed in docker registries :man_shrugging:

File: ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile (hunk 1 above)
Comment (Itxaka) [reply ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile:93]: wtf.

So how is this working on master then? is it just ignoring it? Are they not allowed or simply ignored?

Who the fuck thought that not allowing capital letters was a good idea???? Is this the 80's with their short-sighted tech ideas????

Comment (davidcassany) [reply ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile:93]: > Who the fuck thought that not allowing capital letters was a good idea???? Is this the 80's with their short-sighted tech ideas????

I guess it was the same guy that decided to use some sort of URI for image references but without following regular URI separators and syntax.

> So how is this working on master then? is it just ignoring it? Are they not allowed or simply ignored?

On master it is already without capital letters, I guess there was a post fix for that later on that we might have missed to backport.

Comment (Itxaka) [reply ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile]: found the fix, backported, good catch @davidcassany !

**Review by davidcassany — APPROVED** [review]
LGTM

**Review by pablochacin — APPROVED** [review]
LGTM

**Pull request merged**